src/bindings/js/node/lib/addon.ts

```diff
@@ -0,0 +1,125 @@
+interface Core {
+  compileModel(): CompiledModel;
+  readModelAsync(): Promise<CompiledModel>;
+  readModel(modelPath: string, binPath?: string): Model;
+}
+interface CoreConstructor {
+  new(): Core;
+}
+
+interface Model {
+  outputs: Output[];
+  inputs: Output[];
+
+  output(nameOrId?: string | number): Output;
+  getName(): string;
+}
+
+interface CompiledModel {
+  outputs: Output[];
+  inputs: Output[];
+  output(nameOrId?: string | number): Output;
+  input(nameOrId?: string | number): Output;
+  createInferRequest(): InferRequest;
+}
+
+interface Tensor {
+  // FIXME: now its only Float32Array
+  data: number[];
+  getPrecision(): element;
+  // FIXME: change method return type, when remove Shape
+  getShape(): Shape;
+  // FIXME: now it returns Float32Array
+  getData(): number[];
+}
+interface TensorConstructor {
+  // FIXME: does empty constructor use?
+  new(): Tensor;
+  // FIXME: now tensorData can be only Float32Array
+  new(type?: element, shape?: number[], tensorData?: number[]): Tensor;
+}
+
+interface Shape {
+  data: number[];
+  getData(): number[];
+  shapeSize(): number;
+  getDim(): number;
+}
+interface ShapeConstructor {
+  // FIXME: does empty constructor use?
+  new(): Shape;
+  // FIXME: now tensorData can be only Float32Array
+  new(dimensions: number, data: number[]): Shape;
+}
+
+interface InferRequest {
+  // FIXME: are we going to add index parameter for this method?
+  getOutputTensor(): Tensor;
+  getOutputTensors(): Tensor[];
+  getTensor(output: Output): Tensor;
+  infer(inputData?: { [inputName: string]: Tensor }): void;
+  setInputTensor(tensor: Tensor): void;
+}
+
+interface Output {
+  anyName: string;
+  shape: number[];
+  // Constructor isn't available from JS side
+  // constructor();
+  toString(): string;
+  getAnyName(): string;
+  getShape(): Shape;
+  // FIXME: These methods are not available from JS side for some reason
```

Comment:
Output class represents both C++ Output<const ov::Node> and Output<ov::Node>
`setNames` and `getNames` are only available for Output object that holds Output<ov::Node>, for e.g. in Model class outputs are of class Output<ov::Node>.
We can discuss a different behavior of this this class on JS side as it may be confusing.